src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java

```diff
@@ -0,0 +1,773 @@
+     *   - PUBLISHER: ArXiv-issued DOIs give 'ArXiv' as entry publisher. While this can be true, prefer using one from external sources,
+     *      if applicable
+     *   - KEY_FIELD: Usually, the KEY_FIELD retrieved from user-assigned DOIs are 'nicer' (instead of a DOI link, it's usually contains one author and the year)
+     * */
+    private static final Set<Field> CHOSEN_MANUAL_DOI_FIELDS = Set.of(StandardField.DOI, StandardField.PUBLISHER, InternalField.KEY_FIELD);
+
+    private static final Map<String, String> ARXIV_KEYWORDS_WITH_COMMA_REPLACEMENTS = ImmutableMap.of(
+            "Computational Engineering, Finance, and Science", "Computational Engineering / Finance / Science",
+            "Distributed, Parallel, and Cluster Computing", "Distributed / Parallel / Cluster Computing");
+
+    private final ArXiv arXiv;
+    private final DoiFetcher doiFetcher;
+    private final ImportFormatPreferences importFormatPreferences;
+
+    public ArXivFetcher(ImportFormatPreferences importFormatPreferences) {
+        this.arXiv = new ArXiv(importFormatPreferences);
+        this.doiFetcher = new DoiFetcher(importFormatPreferences);
+        this.importFormatPreferences = importFormatPreferences;
+    }
+
+    /**
+     * NOTE: with this constructor, one can disable the additional search for more metadata (with related DOI) by passing
+     * a NULL DoiFetcher.
+     */
+    public ArXivFetcher(ImportFormatPreferences importFormatPreferences, DoiFetcher doiFetcher) {
+        this.arXiv = new ArXiv(importFormatPreferences);
+        this.doiFetcher = doiFetcher;
+        this.importFormatPreferences = importFormatPreferences;
+    }
+
+    @Override
+    public Optional<URL> findFullText(BibEntry entry) throws IOException {
+        return arXiv.findFullText(entry);
+    }
+
+    @Override
+    public TrustLevel getTrustLevel() {
+        return arXiv.getTrustLevel();
+    }
+
+    @Override
+    public String getName() {
+        return arXiv.getName();
+    }
+
+    @Override
+    public Optional<HelpFile> getHelpPage() {
+        return arXiv.getHelpPage();
+    }
+
+    /**
+     * Remove duplicate values on "KEYWORD" field, if any.
+     *
+     * @param bibEntry A BibEntry to modify
+     */
+    private void adaptKeywordsFrom(BibEntry bibEntry) {
+        Optional<String> allKeywords = bibEntry.getField(StandardField.KEYWORDS);
+        if (allKeywords.isPresent()) {
+            // See https://github.com/JabRef/jabref/pull/9170/#issuecomment-1266042981
+            for (Map.Entry<String, String> entry : ARXIV_KEYWORDS_WITH_COMMA_REPLACEMENTS.entrySet()) {
+                allKeywords = Optional.of(allKeywords.get().replaceAll(entry.getKey(), entry.getValue()));
+            }
+
+            String filteredKeywords = KeywordList.merge(allKeywords.get(), "", importFormatPreferences.getKeywordSeparator()).toString();
+            bibEntry.setField(StandardField.KEYWORDS, filteredKeywords);
```

Comment:
Hummm... I don't see a way to use `KeywordList.replaceAll()` in this context of replacing keywords that contain comma. We would need to first get the KeywordList from `BibEntry.getKeywords(Keyword, Character)`, but this might generate a wrong separation of keywords, as some of those keywords (originated from ArXiv-issued DOIs) contain commas themselves. As this is the exception, I think a string substitution is needed here.

After doing that, however, we could create a KeywordList from this modified string with `KeywordList.parse(keywordString, delimiter)`, retrieve a KeywordList with distinct keywords with `KeywordList.stream().distinct()`, and then modify the BibEntry's keyword field with `BibEntry.putKeywords()`. I think I'd rather this approach than calling a `KeywordList.merge()` with an empty string as a parameter...

Reply:
@koppor  Keyword merge is the best way here, it merges all to a set

Reply:
Well, if there are no further objections to this approach, I'll be leaving this approach.